operators/pkg/controller/elasticsearch/driver/esstate.go

```diff
@@ -13,19 +13,26 @@ import (
 	"github.com/elastic/cloud-on-k8s/operators/pkg/utils/stringsutil"
 )
 
+// ESState gives information about Elasticsearch current status.
 type ESState interface {
+	// NodesInCluster returns true if the given nodes exist in the Elasticsearch cluster.
 	NodesInCluster(nodeNames []string) (bool, error)
+	// ShardAllocationsEnabled returns true if shards allocation are enabled in the cluster.
 	ShardAllocationsEnabled() (bool, error)
+	// GreenHealth returns true if the cluster health is currently green.
 	GreenHealth() (bool, error)
 }
 
+// LazyESState requests Elasticsearch for the requested information only once, at first call.
+// It is "lazy" in the sense it only calls Elasticsearch if required, and does not pre-populate the state.
 type LazyESState struct {
```

Comment:
It feels to me that lazy doesn't tell the full story (the part about cashing the result), but I don't have anything great to use instead.

Reply:
Yes, I agree. At least it's specified in the comment though.

Reply:
`MemoizingESState` memoization implies laziness and caching, not sure if that is a very intuitive name though...

Reply:
I had to Google it, it seems to match perfectly! Not very intuitive but at least you learn new vocabulary when reading the code? 😄 
I'm 👍 switching to that.